src/actions/filter-criteria-actions.js

```diff
@@ -0,0 +1,117 @@
+/**
+ * Copyright 2024 OpenStack Foundation
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+import T from "i18n-react/dist/i18n-react";
+import {
+    getRequest,
+    putRequest,
+    postRequest,
+    deleteRequest,
+    createAction,
+    stopLoading,
+    startLoading,    
+    authErrorHandler,
+    escapeFilterValue,
+    fetchResponseHandler,
+    fetchErrorHandler
+} from "openstack-uicore-foundation/lib/utils/actions";
+import {getAccessTokenSafely} from '../utils/methods';
+
+export const FILTER_CRITERIA_ADDED = 'FILTER_CRITERIA_ADDED';
+export const RECEIVE_FILTER_CRITERIAS = 'RECEIVE_FILTER_CRITERIAS';
+export const FILTER_CRITERIA_DELETED = 'FILTER_CRITERIA_DELETED';
+
+export const saveFilterCriteria = (filterCriteria) => async (dispatch, getState) => {
+    
+    const accessToken = await getAccessTokenSafely();    
+
+    const params = {
+        access_token: accessToken
+    };
+
+    dispatch(startLoading());
+
+    return postRequest(
+        null,
+        createAction(FILTER_CRITERIA_ADDED),
+        `${window.PERSIST_FILTER_CRITERIA_API}/api/v1/filter-criterias`,
+        filterCriteria,
+        authErrorHandler
+    )(params)(dispatch)
+        .then(() => {
+            dispatch(stopLoading());
+        });
+}
+
+export const getFilterCriterias = (context) => async (dispatch, getState) => {    
```

Comment:
what is the purpoise of this action ?
the get filter criteria should be done by the  sleect filter criteria component that is all